src/rust/dbv2/migrations/2024-01-17-184459_coingecko/up.sql

```diff
@@ -0,0 +1,192 @@
+-- Your SQL goes here
+
+-- Replace old utility functions to use api.coins instead of aggregator.coins to allow the functions to be used by web_anon
+CREATE VIEW api.coins AS SELECT * FROM aggregator.coins;
+
+
+CREATE OR REPLACE FUNCTION integer_price_to_quote_indivisible_subunits(market_id numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT ($2 * tick_size * POW(10,COALESCE(base.decimals, 0))) / lot_size
+    FROM market_registration_events
+    LEFT JOIN api.coins base
+    ON base_account_address = base."address" AND base_module_name = base.module AND base_struct_name = base.struct
+    INNER JOIN api.coins quote
+    ON quote_account_address = quote."address" AND quote_module_name = quote.module AND quote_struct_name = quote.struct
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION quote_indivisible_subunits_to_integer_price(market_id numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT ($2 * POW(10,COALESCE(base.decimals, 0)) * lot_size) / tick_size
+    FROM market_registration_events
+    LEFT JOIN api.coins base
+    ON base_account_address = base."address" AND base_module_name = base.module AND base_struct_name = base.struct
+    INNER JOIN api.coins quote
+    ON quote_account_address = quote."address" AND quote_module_name = quote.module AND quote_struct_name = quote.struct
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+
+CREATE OR REPLACE FUNCTION size_to_base_indivisible_subunits(market_id numeric, "size" numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT $2 * lot_size
+    FROM market_registration_events
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+
+CREATE OR REPLACE FUNCTION size_and_price_to_quote_indivisible_subunits(market_id numeric, "size" numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT $2 * $3 * tick_size
+    FROM market_registration_events
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
```

Comment:
I understand you are swapping out `aggregator` for `api` schema in the first two functions, but are you altering these at all?

Reply:
Noting that we covered this offline post-merge and established that indeed this was not needed, but did not affect functionality because of the `CREATE OR REPLACE` statement